plasma_cash/client/history.py

```diff
@@ -0,0 +1,37 @@
+import rlp
+from rlp.sedes import big_endian_int, binary, CountableList
+
+from .exceptions import TxHistoryNotFoundException
+
+
+class History(rlp.Serializable):
+    EMPTY_TX = b'\x00' * 32
+
+    fields = [
+        ('deposit_tx_blk_num', big_endian_int),
+        ('latest_tx_blk_num', big_endian_int),
+        ('latest_blk_num', big_endian_int),
+        ('tx_history', CountableList(binary)),
+        ('proofs', CountableList(binary))
+    ]
+
+    def __init__(self, deposit_tx_blk_num, tx, proof):
+        self.deposit_tx_blk_num = deposit_tx_blk_num
+        self.latest_tx_blk_num = deposit_tx_blk_num
+        self.latest_blk_num = deposit_tx_blk_num
+        self.tx_history = [tx]
+        self.proofs = [proof]
+
+    def update_tx_history(self, tx, proof):
```

Comment:
Instead of re-org concern, I have another concern on that, this implementation requires the consumer to update in the correct order. However, usually, a notification (pub-sub) system would promise delivery over the promising ordering. It's better we can still save the history even if it's not in the right order.

An alternative here could be having an extra `blk_num_list` that saves extra `blk_num` of the tx_history and proofs.
`blk_num` could be passed in as parameter of the function. By doing so, then we don't need to relying on list ordering to know the `blk_num`